Userland/Libraries/LibWeb/HTML/EventLoop/EventLoop.cpp

```diff
@@ -341,4 +367,51 @@ void EventLoop::unregister_environment_settings_object(Badge<EnvironmentSettings
     VERIFY(did_remove);
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#same-loop-windows
+NonnullRefPtrVector<Window> EventLoop::same_loop_windows() const
+{
+    NonnullRefPtrVector<Window> windows;
+    for (auto& document : documents_in_this_event_loop())
+        windows.append(document.window());
+    return windows;
+}
+
+// https://w3c.github.io/hr-time/#dfn-unsafe-shared-current-time
+double EventLoop::unsafe_shared_current_time() const
```

Comment:
Isn't a time_t more appropriate here?

Reply:
The spec uses `DOMHighResTimeStamp` almost everywhere, which is typedef'd to double, but it doesn't define what type `unsafe shared current time` is.